app/Http/Controllers/QuestionController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Validator;
+use App\Question;
+use App\QuestionOption;
+use Illuminate\Http\Request;
+use App\Http\Resources\Question as QuestionResource;
+
+class QuestionController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $questions = Question::all();
+
+        return QuestionResource::collection($questions);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        Validator::make($request->all(), [
+            'topic_id' => 'required|exists:topics,id',
+            'question' => 'required|string|max:65000',
+            'options' => 'required|array',
+        ])->validate();
+      
+        $question = Question::create([
+            'topic_id' => $request->topic_id,
+            'type' => Question::TYPE_ONE_CHOICE,
+            'question' => $request->question,
+        ]);
+        $options = [];
+        $timeNow = now();
+        
+        foreach ($request->options as $option) {
+            $options[] = [
+                'question_id' => $question->id,
+                'option' => $option['option'],
+                'is_right' => $option['is_right'],
+                'created_at' => $timeNow,
+                'updated_at' => $timeNow,
+            ];
+        }
+
+        Validator::make($options, [
```

Comment:
Move this validation into the validation above.

Reply:
There is no sense in the validation after you have tried to interact with the data from the request.